core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -249,6 +248,7 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
         /* register broker metrics */
         registerStats()
 
+        brokerState.newState(RunningAsBroker)
```

Comment:
Btw, one issue that I noticed when doing some other work is that this will override `RunningAsController` even though it probably should not (it was already doing that before this change).